datashader/transfer_functions/__init__.py

```diff
@@ -356,18 +356,16 @@ def _colorize(agg, color_key, how, span, min_alpha, name):
     # if span is provided, use it, otherwise produce it a span based off the
     # min/max of the data
     if span is None:
-        span = [np.nanmin(total).item(), np.nanmax(total).item()]
-        offset = span[0]
-        if how != 'eq_hist':
-            span = _normalize_interpolate_how(how)([0, span[1] - span[0]], 0)
-        # If there are elements that are zero or lower, they should be fully
-        # transparent and masked out
-        if offset <= 0:
+        # Currently masks out zero or negative values, but will need fixing         
+        offset = np.nanmin(total)
+        if offset <= 0 and total.dtype.kind == 'u':
```

Comment:
```suggestion
        if offset == 0 and total.dtype.kind == 'u':
```

(a unit can't be less than zero, right?)